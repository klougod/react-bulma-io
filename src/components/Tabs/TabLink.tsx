import React from 'react'
import PropTypes from 'prop-types'

export const TabLink = ({ isActive, text, id, changeTab }: any) => {
  return (
    <li className={isActive ? 'is-active' : ''}>
      {/* eslint-disable-next-line */}
      <a
        id={id}
        className='is-size-4-tablet is-size-6-mobile'
        style={{ padding: '0.5em 0.2em !important' }}
        onClick={changeTab}
      >
        {text}
      </a>
    </li>
  )
}

TabLink.defaultProps = {
  className: '',
  id: '',
  text: '',
  isActive: false,
  changeTab: () => {}
}

TabLink.propTypes = {
  className: PropTypes.string,
  id: PropTypes.string,
  text: PropTypes.string,
  isActive: PropTypes.bool,
  changeTab: PropTypes.func,
  children: PropTypes.node
}

export default TabLink
