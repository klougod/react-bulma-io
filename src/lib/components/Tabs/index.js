export { default as Tabs } from './Tabs'
export { default as TabLink } from './TabLink'
