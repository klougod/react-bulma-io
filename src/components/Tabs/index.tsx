import React, { Fragment, useState, ReactElement, MouseEvent } from 'react'
import PropTypes from 'prop-types'

import TabLink from './TabLink'

const Tabs = ({ children, className }: any) => {
  const [arrayChildren, setChildren] = useState(
    React.Children.toArray(children)
  )

  const changeTab = (e: MouseEvent<HTMLButtonElement>) => {
    const key = (e.target as HTMLButtonElement).id
    setChildren(
      arrayChildren.map((child: ReactElement) =>
        React.cloneElement(child, {
          ...child.props,
          isActive: child.key === key
        })
      )
    )
  }

  return (
    <Fragment>
      <div className={`tabs ${className}`}>
        <ul>
          {arrayChildren.map((child: ReactElement, i) => (
            <TabLink
              id={`.${i}`}
              isActive={child.props.isActive}
              changeTab={changeTab}
              key={i}
              text={child.props.tabName}
            />
          ))}
        </ul>
      </div>
      <div
        style={{
          borderRadius: '0px',
          marginLeft: '1px',
          width: 'calc(100% - 2px)'
        }}
        className='column is-12 box'
      >
        {arrayChildren.filter((c: ReactElement) => c.props.isActive)}
      </div>
    </Fragment>
  )
}

Tabs.defaultProps = {
  className: ''
}

Tabs.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node
}

export { TabLink, Tabs }
export default Tabs
