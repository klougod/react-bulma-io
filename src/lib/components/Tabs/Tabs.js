import React, { Fragment, useState } from 'react'
import PropTypes from 'prop-types'
import TabLink from './TabLink'


export const Tabs = ({ children, className }) => {

  const [arrayChildren, setChildren] = useState(React.Children.toArray(children))

  const changeTab = e => {
    const key = e.target.id
    setChildren(arrayChildren.map((child) => (
      React.cloneElement(child, { ...child.props, isActive: child.key === key })
    )))
  }

  return (
    <Fragment>
      <div className={`tabs ${className}`}>
        <ul>
          {
            arrayChildren.map((child, i) => (
              <TabLink id={`tab-link-${i}`} isActive={child.props.isActive} changeTab={changeTab}
                key={i} text={child.props.tabName} />
            ))
          }
        </ul>
      </div>
      <div style={{ 'borderRadius': '0px', 'marginLeft': '1px', 'width': 'calc(100% - 2px) !important' }}
        className="column is-12 box" >
        {arrayChildren.filter(c => c.props.isActive)}
      </div>
    </Fragment>
  );
}

Tabs.defaultProps = {
  className: ''
}

Tabs.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired
}

export default Tabs