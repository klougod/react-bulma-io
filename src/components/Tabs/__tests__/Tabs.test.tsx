import React from 'react'
import Enzyme from 'enzyme'

import Tabs from '../'
import TabLink from '../TabLink'

describe('Tabs component test', function () {
  it('should render without crash with wrapper', function () {
    const tabsMock = Enzyme.shallow(<Tabs />)
    expect(tabsMock).toMatchSnapshot()
  })

  it('should render without crash with wrapper', function () {
    const tabsMockProps = { noWrapper: true }
    const tabsMock = Enzyme.shallow(
      <Tabs {...tabsMockProps}>
        <TabLink />
        <TabLink />
        <TabLink />
      </Tabs>
    )
    expect(tabsMock).toMatchSnapshot()
  })
})
