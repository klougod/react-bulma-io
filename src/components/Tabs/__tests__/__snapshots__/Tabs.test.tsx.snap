// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs component test should render without crash with wrapper 1`] = `
<Fragment>
  <div
    className="tabs "
  >
    <ul />
  </div>
  <div
    className="column is-12 box"
    style={
      Object {
        "borderRadius": "0px",
        "marginLeft": "1px",
        "width": "calc(100% - 2px)",
      }
    }
  />
</Fragment>
`;

exports[`Tabs component test should render without crash with wrapper 2`] = `
<Fragment>
  <div
    className="tabs "
  >
    <ul>
      <Component
        changeTab={[Function]}
        className=""
        id=".0"
        isActive={false}
        key="0"
        text=""
      />
      <Component
        changeTab={[Function]}
        className=""
        id=".1"
        isActive={false}
        key="1"
        text=""
      />
      <Component
        changeTab={[Function]}
        className=""
        id=".2"
        isActive={false}
        key="2"
        text=""
      />
    </ul>
  </div>
  <div
    className="column is-12 box"
    style={
      Object {
        "borderRadius": "0px",
        "marginLeft": "1px",
        "width": "calc(100% - 2px)",
      }
    }
  />
</Fragment>
`;
